### DISCUSSION TITLE
RabbitMQ health check failed

### DISCUSSION THREAD

**Discussion by niraj-ranjan-comprinno** [rabbitmq/rabbitmq-java-client#1068] [General]
### Describe the bug

In last recent days, We are getting rabbitMq health check failed.
Then we are getting Connection Refused error.

This error is coming randomly. Not occuring on starup the application.

### Reproduction steps

@Component
public class CloudAMQPHealthIndicator extends AbstractHealthIndicator {

	final static ConnectionFactory factory = new ConnectionFactory();
	
    @Override
    protected void doHealthCheck(Health.Builder builder) throws Exception {
        // Use the builder to build the health status details that should be reported.
        // If you throw an exception, the status will be DOWN with the exception message.
		String cloudAmqpUrl = CustomProperties.getProperty("CLOUDAMQP_URL");
        if (cloudAmqpUrl == null || cloudAmqpUrl.trim().length() == 0) {
        	builder.up().withDetail("description", "Cloudamqp url not found.");
        	throw new BusinessException("Cloudamqp url not found.");
		}
		
		Connection connection = null;
		try {
		    factory.setUri(CustomProperties.getProperty("CLOUDAMQP_URL"));
			
			connection = factory.newConnection();
			builder.up()
                .withDetail("description", "UP");
		} catch (Exception e) {
			builder.down()
                .withDetail("description", e.getMessage());
			throw e;
		} finally {
			try {
				if (connection != null) {
					connection.close();
				}
			} catch (Exception e) {
				e.printStackTrace();
			}
		}
		
    }

### Expected behavior

Long lived connection must be maintain.

**Comment by acogoluegnes** [comment]
Please provide more information: broker, Erlang, and Java client versions, broker and client logs. We need more context to help.

You write "in last recent days", does this mean it was working correctly before and you're getting these errors now? If so, what has changed?

**Reply by niraj-ranjan-comprinno** [reply]
Yes now

**Reply by niraj-ranjan-comprinno** [reply]
<dependency>
			<groupId>com.rabbitmq</groupId>
			<artifactId>amqp-client</artifactId>
			<version>2.8.2</version>
		</dependency>

**Reply by niraj-ranjan-comprinno** [reply]
@acogoluegnes

**Reply by acogoluegnes** [reply]
So what did you change (from what to what)?

2.8.2 is more than 10 years old, you need to upgrade.

**Reply by Alan7481** [reply]
thank you

**Comment by michaelklishin** [comment]
RabbitMQ core team are not responsible for CloudAMQP clusters (you try to connect using `CLOUDAMQP_URL`). Please contact CloudAMQP support.

Long-lived client connections will fail from time to time, this is an inevitable reality of distributed systems. This client has had [a connection recovery mechanism](https://rabbitmq.com/api-guide.html#recovery) for about a decade now for that very reason.

**Comment by acogoluegnes** [comment]
BTW, you're creating a new connection and closing it for every call to `doHealthCheck`, so this is not consistent with the expectation you're mentioning (_Long lived connection must be maintain_).
